File: Chapter12/Android_Toast/mytoast.cpp
#include "mytoast.h"

MyToast::MyToast(QObject *parent) : QObject(parent) {}

void MyToast::callToast(QString message)
{
	#ifdef Q_OS_ANDROID
	QtAndroid::runOnAndroidThread([=]
	{
		QAndroidJniObject::callStaticMethod<void>("org/qtproject/example/MyToast",
		"notify", "(Landroid/content/Context;Ljava/lang/String;)V",
		QtAndroid::androidActivity().object(),
		QAndroidJniObject::fromString(message).object<jstring>());
	});
	#endif
}

File: Chapter 12/Android_Toast/android/src/org/qtproject/example/MyToast.java
package org.qtproject.example;

import android.content.Context;
import android.widget.Toast;

public class MyToast extends org.qtproject.qt5.android.bindings.QtActivity
{
    public MyToast()
    {
    }

    public static void notify(Context context, String message)
    {
	int duration = Toast.LENGTH_SHORT;
	Toast toast = Toast.makeText(context, message, duration);
	toast.show();
    }
}
